codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/TestHelpers.kt

```diff
@@ -105,9 +105,8 @@ fun StructureShape.renderWithModelBuilder(
     model: Model,
     symbolProvider: RustSymbolProvider,
     writer: RustWriter,
-    forWhom: CodegenTarget = CodegenTarget.CLIENT,
```

Comment:
This change is going to result in the errors being generated incorrectly for server tests.

Perhaps we should move this function into `codegen-client` and make an equivalent in `codegen-server`.